rpc-interface/src/blockchain.rs

```diff
@@ -97,10 +99,36 @@ pub trait BlockchainInterface {
         &mut self,
         address: Address,
         include_stakers: Option<bool>,
-    ) -> Result<Validator, Self::Error>;
+    ) -> Result<BlockchainState<Validator>, Self::Error>;
 
-    async fn get_staker_by_address(&mut self, address: Address) -> Result<Staker, Self::Error>;
+    async fn get_staker_by_address(
+        &mut self,
+        address: Address,
+    ) -> Result<BlockchainState<Staker>, Self::Error>;
+
+    #[stream]
+    async fn head_subscribe(
+        &mut self,
+        include_transactions: Option<bool>,
+    ) -> Result<BoxStream<'static, Block>, Self::Error>;
+
+    #[stream]
+    async fn head_hash_subscribe(&mut self)
+        -> Result<BoxStream<'static, Blake2bHash>, Self::Error>;
 
     #[stream]
-    async fn head_subscribe(&mut self) -> Result<BoxStream<'static, Blake2bHash>, Self::Error>;
+    async fn election_validator_subscribe(
+        &mut self,
+        address: Address,
+    ) -> Result<BoxStream<'static, BlockchainState<Validator>>, Self::Error>;
+
+    #[stream]
+    async fn logs_subscribe(&mut self) -> Result<BoxStream<'static, BlockLog>, Self::Error>;
```

Comment:
```suggestion
    async fn logs_subscribe(&mut self) -> Result<BoxStream<'static, Log>, Self::Error>;
```
One more, shouldn't this be a `Log` instead of a `BlockLog`?